core/shared/platform/common/posix/posix_file.c

```diff
@@ -52,6 +54,17 @@
 #define CONFIG_HAS_O_SYNC
 #endif
 
+// Provides a dummy definition which is not actually used,
+// but is required to compile.
+#if WASM_ENABLE_LIBC_WASI == 0
+__wasi_errno_t
+convert_errno(int error)
```

Comment:
Previously we were only compiling this file if wasi libc was enabled - has that changed? I think that would make more sense since all the functions in this file are only used when wasi libc is enabled. It also means we don't need to add the stub which never gets used.

Reply:
In this PR, I'm trying to use `os_file_handle ` to let the `os_memmap` support map memory from file, and then use the ` os_get_invalid_handle()` as invalid file handle.

But the `os_get_invalid_handle()` introduced the usage of `convert_errno()`. Add '-ffunction-section` to compile flag is another approach  but we should not depend on specifal compiler flag .

Reply:
The `convert_errorno` has been implemented in `core/shared/platform/common/libc-util/libc_errno.c/h`:
  https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/platform/common/libc-util/libc_errno.c#L9-L10
No need to add it again here. I think maybe we should add `libc_error.c` into file list in Nuttx's makefile.

BTW, had better not change `shared_platform.cmake` of windows platform.

Reply:
This issue is exists on all platform, if [posix | win]_file.c compiled，then we must add libc_errno.c to build system.